apps/ledger-live-mobile/src/components/FabActions/modals/ZeroBalanceDisabledModalContent.tsx

```diff
@@ -20,41 +20,39 @@ function ZeroBalanceDisabledModalContent({
 
   const actionCurrency = account ? getAccountCurrency(account) : currency;
 
-  const goToBuy = useCallback(
-    () =>
-      navigation.navigate(NavigatorName.Exchange, {
-        screen: ScreenName.ExchangeBuy,
-        params: {
-          defaultCurrencyId: actionCurrency.id,
-          defaultAccountId: account && account.id,
-        },
-      }),
-    [],
-  );
+  const goToBuy = useCallback(() => {
+    navigation.navigate(NavigatorName.Exchange, {
+      screen: ScreenName.ExchangeBuy,
+      params: {
+        defaultCurrencyId: actionCurrency?.id,
+        defaultAccountId: account?.id,
+      },
+    });
+    onClose();
+  }, [account?.id, actionCurrency?.id, navigation, onClose]);
 
-  const goToReceive = useCallback(
-    () =>
-      navigation.navigate(NavigatorName.ReceiveFunds, {
-        screen: account
-          ? ScreenName.ReceiveConnectDevice
-          : ScreenName.ReceiveSelectAccount,
-        params: {
-          selectedCurrency: actionCurrency,
-          accountId: account && account.id,
-        },
-      }),
-    [navigation],
-  );
+  const goToReceive = useCallback(() => {
+    navigation.navigate(NavigatorName.ReceiveFunds, {
+      screen: account
+        ? ScreenName.ReceiveConfirmation
+        : ScreenName.ReceiveSelectAccount,
+      params: {
+        selectedCurrency: actionCurrency,
```

Comment:
I guess we should also add a param called `currency` for the ReceiveConfirmation screen
```suggestion
        selectedCurrency: actionCurrency,
        currency: actionCurrency,
```